KubeArmor/enforcer/bpflsm/enforcer.go

```diff
@@ -207,13 +219,137 @@ func NewBPFEnforcer(node tp.Node, pinpath string, logger *fd.Feeder) (*BPFEnforc
 		}
 	}
 
+	be.Events, err = ringbuf.NewReader(be.obj.Events)
+	if err != nil {
+		be.Logger.Errf("opening ringbuf reader: %s", err)
+		return be, err
+	}
+	be.EventsChannel = make(chan []byte, mon.SyscallChannelSize)
+
+	go be.TraceEvents()
+
 	if cfg.GlobalCfg.HostPolicy {
 		be.AddHostToMap()
 	}
 
 	return be, nil
 }
 
+type eventBPF struct {
+	Ts uint64
+
+	PidID uint32
+	MntID uint32
+
+	HostPPID uint32
+	HostPID  uint32
+
+	PPID uint32
+	PID  uint32
+	UID  uint32
+
+	EventID int32
+	Retval  int64
+
+	Comm [16]byte
+
+	Data InnerKey
+}
+
+func (be *BPFEnforcer) TraceEvents() {
+	if be.Events == nil {
+		be.Logger.Err("ringbuf reader is nil, exiting trace events")
+	}
+	be.Logger.Print("Starting TraceEvents from BPF LSM Enforcer")
+	go func() {
+		for {
+			be.Logger.Warnf("Waiting for event from BPF LSM Enforcer")
+			record, err := be.Events.Read()
+			if err != nil {
+				if errors.Is(err, ringbuf.ErrClosed) {
+					// This should only happen when we call DestroyMonitor while terminating the process.
+					// Adding a Warn just in case it happens at runtime, to help debug
+					be.Logger.Warnf("Ring Buffer closed, exiting TraceEvents %s", err.Error())
+					return
+				}
+				be.Logger.Warnf("Ringbug error reading %s", err.Error())
+				continue
+			}
+			be.Logger.Warnf("Received Event from BPF LSM Enforcer")
+
+			be.EventsChannel <- record.RawSample
+
+		}
+	}()
+
+	for {
+		select {
+
+		case dataRaw := <-be.EventsChannel:
+			// if !valid {
+			// 	continue
+			// }
+
+			var event eventBPF
+
+			if err := binary.Read(bytes.NewBuffer(dataRaw), binary.LittleEndian, &event); err != nil {
+				log.Printf("parsing ringbuf event: %s", err)
+				continue
+			}
+
+			containerID := ""
+
+			if event.PidID != 0 && event.MntID != 0 {
+				containerID = be.Monitor.LookupContainerID(event.PidID, event.MntID, event.HostPPID, event.HostPID)
+			}
+
+			log := be.Monitor.BuildLogBase(event.EventID, mon.ContextCombined{
+				ContainerID: containerID,
+				ContextSys: mon.SyscallContext{
+					PID:  event.PID,
+					PPID: event.PPID,
+					UID:  event.UID,
+
+					HostPID:  event.HostPID,
+					HostPPID: event.HostPPID,
+				},
+			})
+			// Add Switch Case Code for Event ID based on File/Process/Network
+
+			switch event.EventID {
+
+			case File_Open, File_Permission, File_Mknod, File_Mkdir, File_Rmdir, File_Unlink, File_Symlink, File_Link, File_Rename, File_Chmod, File_Truncate:
+				log.Operation = "File"
+				log.Resource = string(event.Data.Path[:])
+				log.Enforcer = "BPFLSM"
+				log.Result = "Permission denied"
+				log.Data = "syscall=" + getSyscallName(int32(event.EventID))
```

Comment:
```suggestion
				log.Data = "lsm=" + getSyscallName(int32(event.EventID))
```

Let's use the keyword lsm in Data, since the telemetry is not from a syscall but lsm hook